ckan/lib/helpers.py

```diff
@@ -152,14 +152,37 @@ def redirect_to(*args, **kw):
 
         toolkit.redirect_to('dataset_read', id='changed')
 
+    If given a single string as argument, this redirects without url parsing
+
+        toolkit.redirect_to('http://example.com')
+        toolkit.redirect_to('/dataset')
+        toolkit.redirect_to('/some/other/path')
+
     '''
     if are_there_flash_messages():
         kw['__no_cache__'] = True
 
     # Routes router doesn't like unicode args
     uargs = map(lambda arg: str(arg) if isinstance(arg, unicode) else arg,
                 args)
-    _url = url_for(*uargs, **kw)
+
+    # Repoze.who redirects with single strings
+    # which need to be parsed by url_for
+    exempt_urls = ['/user/logout', '/user/logged_out_redirect',
+                   '/user/logged_in']
```

Comment:
It might be detectable by using __ckan_no_root parameter, since the only place it is used is in user controller during logins like [here](https://github.com/ckan/ckan/blob/master/ckan/controllers/user.py#L400). But I'll test it tomorrow if it helps on update the PR if it does.